DotBot.py

```diff
@@ -9,43 +9,47 @@
 # setup logging
 import logging.config
 import LoggingConfig
+
 logging.config.dictConfig(LoggingConfig.LOGGINGCONFIG)
 log = logging.getLogger('DotBot')
 
+
 class DotBot(Wireless.WirelessDevice):
     '''
     A single DotBot.
     '''
 
-    RETRY_TIMEOUT_S               = 1
+    RETRY_TIMEOUT_S             = 1
 
     def __init__(self, dotBotId, x, y, floorplan):
 
         # store params
-        self.dotBotId             = dotBotId
-        self.x                    = x
-        self.y                    = y
-        self.floorplan            = floorplan
+        self.dotBotId           = dotBotId
+        self.x                  = x
+        self.y                  = y
+        self.floorplan          = floorplan
 
-        #=== local variables
+        # === local variables
         # singletons
-        self.simEngine            = SimEngine.SimEngine()
-        self.wireless             = Wireless.Wireless()
+        self.simEngine          = SimEngine.SimEngine()
+        self.wireless           = Wireless.Wireless()
         # current heading and speed
-        self.currentHeading       = 0
-        self.currentSpeed         = 0
+        self.currentHeading     = 0
+        self.currentSpeed       = 0
         # timestamps of when movement starts/stops
-        self.tsMovementStart      = None
-        self.tsMovementStop       = None
+        self.tsMovementStart    = None
+        self.tsMovementStop     = None
         # sequence numbers (to filter out duplicate commands and notifications)
-        self.seqNumCommand        = None
-        self.seqNumNotification   = 0
+        self.seqNumCommand      = None
+        self.seqNumNotification = 0
         # state maintained by internal bump computation
-        self.nextBumpX            = None  # coordinate the DotBot will bump into next
-        self.nextBumpY            = None
-        self.nextBumpTime         = None  # time at which DotBot will bump
+        self.nextBumpX          = None  # coordinate the DotBot will bump into next
+        self.nextBumpY          = None
+        self.nextBumpTime       = None  # time at which DotBot will bump
         # is dotBot a relay
-        self.isRelay              = False
+        self.isRelay            = False
+        # if dotbot has bumped
+        self.bumped             = False
```

Comment:
because we stop when we reach target too (even if no bump), so we call transmit from multiple functions now with different bump status, sometimes its true, sometimes false